src/Compilers/VisualBasic/Portable/Syntax/InternalSyntax/SyntaxToken.vb

```diff
@@ -89,6 +89,13 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
                 Me._trailingTrivia = DirectCast(reader.ReadValue(), GreenNode)
             End Sub
 
+            Private ReadOnly Property IsReausable As Boolean Implements IObjectWritable.IsReusable
+                Get
+                    Return _leadingTrivia IsNot Nothing AndAlso
+                        ShouldCacheTriviaInfo(_leadingTrivia, _trailingTrivia)
```

Comment:
can this method be commented.  this appears to be the only one who can return 'false' for IsReusable.  but it's not clear ot me why. For example, why is trailing trivia not checked?  Whatever the reasons, docs here seem appropriate.

Reply:
`ShouldCacheTriviaInfo` has comments. This is all heuristics. 
Node caching in parser is on a very hot path so the checks need to balance between being useful and what can be done very fast.
There is some logic there, but also some things are done just because it is cheaper to check and it is good enough.


---
In reply to: [181531292](https://github.com/dotnet/roslyn/pull/26156#discussion_r181531292) [](ancestors = 181531292)

Reply:
`ShouldCacheTriviaInfo` does not expect `_leadingTrivia` be `Nothing` since noone calls it with that. It may just not be possible for the `TriviaInfo` not have leading trivia. Perhaps the check is unnecessary.

---
In reply to: [181558869](https://github.com/dotnet/roslyn/pull/26156#discussion_r181558869) [](ancestors = 181558869,181531292)

Reply:
basically, it would be good if all this information was in a comment in the source code :)  otherwise no one will know to look here for this info

Reply:
I mean - there is a comment 20 lines above, - we only now call `ShouldCacheTriviaInfo` here, not even doing a null chack since it appears that leading trivia is never null by construction. And there is an Assert for that effect just above. I do not think there is a lot to document here.


---
In reply to: [181559878](https://github.com/dotnet/roslyn/pull/26156#discussion_r181559878) [](ancestors = 181559878)